aten/src/TH/THTensor.cpp

```diff
@@ -128,6 +133,8 @@ void THTensor_resizeNd(THTensor *self, int nDimension, const int64_t *size, cons
       THStorage_resize(THTensor_getStoragePtr(self), totalSize+self->storage_offset());
     }
   }
+  THTensor_refreshContiguous(self);
+  self->refresh_numel();
```

Comment:
THTensor_setStrideAtDim and friends should be continuously keeping the invariant correct, no? So these calls should not be necessary.